### PR TITLE
GITC-478: Caching github user object.

### PR DESCRIPTION
**Pull request by nutrina**
Have created model to store the serialized github entities. Have also reworked code, mainly in the `git/utils.py` file to make use of this cache.

##### Description

This fixes the caching and conditional requests for the user object from Github API.
For now this only handles the user obejct requests, but this fix can be extended for repo and issue objects as well.

##### Refers/Fixes

GITC-478
This should fix the high amount of rate limit errors in sentry.

##### Testing

This would normally be triggered by going to the profile or dashboard pages for a user.

### CODE REVIEW

File: app/git/models.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import logging
+
+from django.db import models
+
+from economy.models import SuperModel
+
+logger = logging.getLogger(__name__)
+
+
+class GitCache(SuperModel):
```
Comment (ksolo) [inline app/git/models.py:31]: @nutrina can you add tests for this model?

Comment (nutrina) [reply app/git/models.py:31]: @ksolo Yes, I am on it

**Comment by nutrina** [comment]
@ksolo Have added test cases

**Review by ksolo — APPROVED** [review]

**Comment by nutrina** [comment]
@thelostone-mc  I have rebased this to stable

**Review by thelostone-mc — APPROVED** [review]

**Review by alicii93 — COMMENTED** [review]
10x09D1564FB8344A4d0F74975bCed78eB7c7912617

**Pull request merged**